geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/StartLocatorCommand.java

```diff
@@ -139,6 +139,7 @@ public ResultModel startLocator(
 
   }
 
+  @SuppressWarnings("deprecation")
```

Comment:
In general, I would encourage you to not use `@SuppressWarnings("deprecation")`. All that does it hide how much we're using deprecated APIs, whereas I think we need to be aware of how much deprecated code we're calling and eventually change it. If it's hidden we'll forget and not revisit it.

Reply:
our build would fail if we don't suppress it. I believe Jake just put in a check to prevent the compiler to build successfully if we introduced warning. but in this case, we had to use the deprecated variable since "start locator" still needs to support it.

Reply:
Hmm. That is actually really bad if true, because many of the API are not properly type and are raw. So in writing new tests, there would be a real problem.

Reply:
Fix the root problem. We need to stop propagating bad habits and warnings. If the method is deprecated, don't use it.

Reply:
> Fix the root problem. We need to stop propagating bad habits and warnings. If the method is deprecated, don't use it.

@pivotal-jbarrett the code needs to refer to the `ConfigurationProperties.LOAD_CLUSTER_CONFIGURATION_FROM_DIR` directly, hence the warning, are you saying we need to change it to a string literal here?

Reply:
If the use of this property is truly deprecated then it should not be deprecated. If there is some legitimate reason for us of a deprecated property or method then isolate it and suppress that single statement. Suppressing this entire method could mask other issues. Suppression should be at the smallest scope possible.

Reply:
Replace the offending line 197 with a method like this:
```java
  @SuppressWarnings("deprecation")
  private void setLoadClusterConfigurationFromDir(Properties gemfireProperties,
                                                  boolean loadSharedConfigurationFromDirectory) {
    StartMemberUtils.setPropertyIfNotNull(gemfireProperties,
        ConfigurationProperties.LOAD_CLUSTER_CONFIGURATION_FROM_DIR,
        loadSharedConfigurationFromDirectory);
  }
```